File: Algorithm I/7. Bit Manipulation/P136.cpp
// Date: Fri, Oct 1, 2021
// 136. Single Number (https://leetcode.com/problems/single-number/)
// Constraint: linear runtime and constant extra space

int singleNumber(vector<int>& nums) {
	int ans = 0;
	for (int num: nums) {
		ans ^= num;
	}
	return ans;
}

// reference: 
// https://www.hackerearth.com/practice/basic-programming/bit-manipulation/basics-of-bit-manipulation/tutorial/
https://www.geeksforgeeks.org/bits-manipulation-important-tactics/

File: LeetCoding-Challenge/February-2022/P136.java
package leetcodingchallenge.february2022;

/**
 * Date: Feb 15, 2022
 * 136. Single Number
 * https://leetcode.com/problems/single-number/
 */

public class P136 {
    public int singleNumber(int[] nums) {
        int ans = 0;
        for (int num : nums)
            ans ^= num;
        return ans;
    }
}
